Misc/NEWS.d/next/Core_and_Builtins/2024-08-23-21-20-34.gh-issue-123271.xeVViR.rst

```diff
@@ -0,0 +1 @@
+Make :meth:`zip` safe under free-threading.
```

Comment:
```suggestion
Make :func:`zip` thread-safe without the :term:`GIL`.
```

Reply:
Even with this PR the `zip` function is not thread-safe: when iterating over `zip(range(100), range(100))` one can obtain a tuple `(2,3)`. What does PR (hopefully) does is modifying the code so that the interpreter does not crash. For this reason I would like to be careful with the term "thread-safe". I choose not to make `zip` fully thread safe because of the performance impact, see the discussion in https://github.com/python/cpython/issues/120496

Reply:
Thanks for clarifying. The word `safe` is a bit confusing, I guess a lot of other people would read it as `thread-safe`. I think we need to clarify the reason why `zip` isn't safe in `free-threaded` build.